### PR TITLE
[AIRFLOW-566] Add timeout while fetching logs

### PR DESCRIPTION
**Pull request by msumit**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- _https://issues.apache.org/jira/browse/AIRFLOW-566_

Testing Done:
- Manuall tested

`Add request timeout while fetching logs from different machine, so it
does not take forever incase the other machine is down and move on to
next step, which is fetching logs from clouds.`

### CODE REVIEW

**Comment by msumit** [comment]
Landscape and Travis both giving false alerts.. Landscape showing 6 warnings fixed, though they are not part of my diff. Travis run on my repo got succeeded https://travis-ci.org/msumit/airflow/builds/167346507..

**Comment by gwax** [comment]
+1

**Comment by r39132** [comment]
@msumit looks good... wondering if we should make this a configuration variable rather than a magic number?

**Comment by r39132** [comment]
@msumit please respond within 5 days else will close this.

**Comment by msumit** [comment]
@r39132 moved value behind a conf var.. let me know if it looks ok now, so I can rebase..

File: airflow/configuration.py [hunk 1]
```diff
@@ -252,6 +252,10 @@ def run_command(command):
 # privacy.
 demo_mode = False
 
+# The amount of time (in secs) webserver will wait for initial handshake
+# while fetching logs from other worker machine
+handshake_timeout = 5
```
Comment (r39132) [inline airflow/configuration.py]: `handshake_timeout` sounds too vague.

I'd recommend `s3_log_fetch_timeout_secs`.

**Review by r39132 — CHANGES_REQUESTED** [review]
Looks good otherwise.. pls make requested changes and prepare for merge (e.g. squash commits, rebase if needed)

File: airflow/configuration.py (hunk 1 above)
Comment (msumit) [reply airflow/configuration.py]: may be just `log_fetch_timeout_sec` as this timeout only impact when log gets fetched from other machines..

**Comment by r39132** [comment]
+1 I'll go ahead and squash on merge.

**Pull request closed without merging**